### DISCUSSION TITLE
get drop file from q-file?

### DISCUSSION THREAD

**Discussion by atticus-lv** [zauberzeug/nicegui#3041] [Q&A]
### Question

hi there. I wonder if I can use `ui.element('q-file').props('filled label="Drop File Here"')` just like the `ui.upload`(both pick/drop files)
Sometimes the ui.upload is too big to place in a single line, so I want to use [q-file](https://quasar.dev/vue-components/file) for custom upload(drop file almost. I have already try the file picker example).
 However I don't know how to get to callback from this custom element

**Comment by falkoschindler** [comment]
Hi @atticus-lv,

I think Quasar's [QFile](https://quasar.dev/vue-components/file) can only pick files, but can't _upload_ them.

Just recently I found the following solution for another project. It creates a hidden `ui.upload` element with `auto_upload` activated and a button to trigger its "pickFiles" method:

```py
upload = ui.upload(auto_upload=True, on_upload=lambda e: ui.notify(f'{len(e.content.read())} bytes')).classes('hidden')
ui.button('Upload', on_click=lambda: upload.run_method('pickFiles'))
```

Even though you don't see the list of uploaded files, this approach might be suitable for a more minimal UI for uploading a file.

**Reply by atticus-lv** [reply]
hi falkoschindler, thanks for your reply. that is a good solution compared to the file picker example. However, I want to get file list when droping just as the q-file.

**Comment by PoshoDev** [comment]
I managed to get the selected file with the `update:model-value` event. :+1: 

```python
from nicegui import ui

qfile = ui.element('q-file').props('filled label="Drop File Here"')
qfile.on('update:model-value', lambda e: print(f"File: '{e.args.get('name')}'"))

ui.run()
```

What I don't understand is why the _Q-File_ element is not showing the selected file inside of it. I looked at the [documentation](https://quasar.dev/vue-components/file/) and I don't see any special thing I could be missing out. Even adding the `v-model="file"` prop does nothing for me. :thinking:

**Reply by falkoschindler** [reply]
It looks like QFile doesn't allow to change its model value programmatically:
```py
ui.button('Set value', on_click=lambda: qfile.props('model-value="test"'))  # doesn't work
```

But there's a "display-value" prop we can use. Full example:
```py
qfile = ui.element('q-file').props('filled label="Drop File Here"').classes('w-32')
qfile.on('update:model-value', lambda e: qfile.props(f'display-value={e.args.get("name")}'))
```

**Reply by PoshoDev** [reply]
Perfect! This solves everything! Thank you! :)
